okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/UrlSelectorImpl.java

```diff
@@ -64,11 +68,11 @@ private UrlSelectorImpl(ImmutableList<HttpUrl> baseUrls, boolean reshuffle, Dura
         // Assuming that baseUrls is already randomized, start with the first one.
         this.lastBaseUrl = new AtomicReference<>(baseUrls.get(0));
 
-        long coolDownMillis = failedUrlCooldown.toMillis();
+        this.clock = clock;
         this.failedUrls = Caffeine.newBuilder()
+                .executor(MoreExecutors.directExecutor())
                 .maximumSize(baseUrls.size())
-                .expireAfterWrite(coolDownMillis, TimeUnit.MILLISECONDS)
-                .build();
+                .build(key -> clock.instant().plus(failedUrlCooldown));
```

Comment:
This cache looks quite strange to me now - I don't really understand why it's desirable to have a fixed expiry time for every entry?? What if a UrlSelectorImpl isntance lives for a long time?

Reply:
> I don't really understand why it's desirable to have a fixed expiry time for every entry??

Isn't this exactly the meaning of `coolDownMillis`? What are you suggesting as an alternative?

Reply:
Right, we don't want this anymore otherwise we might mark previously failed urls as healthy again. The maximum size should be sufficient to limit the size of this cache.

Reply:
This cache no longer does any expiry. We also limit the size of the cache to the size of the base urls list. What's the concern here?

Reply:
Well in any case this doesn't need to be a cache, turns out. So I've made it a map.